crypto/lhash/lhash.c

```diff
@@ -190,18 +163,8 @@ void *OPENSSL_LH_retrieve(OPENSSL_LHASH *lh, const void *data)
     unsigned long hash;
     OPENSSL_LH_NODE **rn;
 
-    /*-
-     * This should be atomic without tsan.
-     * It's not clear why it was done this way and not elsewhere.
-     */
-    tsan_store((TSAN_QUALIFIER int *)&lh->error, 0);
```

Comment:
We still need the `lh->error = 0;` side effect of this.

This was left using tsan because I'm paranoid :)

Reply:
Having this mutation on read-only access to the LHASH isn't ideal, but I guess it'll probably be put in a write buffer by high performance hardware, and we need it to meet API. As an aside, anyone know if modern hardware does an equality comparison before changing a shared cache line from shared to modified (MESI)?